crates/primitives/src/transaction/tx_type.rs

```diff
@@ -54,6 +57,30 @@ impl TxType {
             TxType::DEPOSIT => false,
         }
     }
+
+    /// Returns the max size for a transaction type.
+    pub const fn max_encoded_tx_length(&self) -> usize {
+        match self {
+            TxType::Legacy => usize::pow(2, 20),  // todo: set limit
+            TxType::EIP2930 => usize::pow(2, 20), // todo: set limit
+            TxType::EIP1559 => usize::pow(2, 20), // todo: set limit
+            TxType::EIP4844 => usize::pow(2, 20),
+            #[cfg(feature = "optimism")]
+            TxType::DEPOSIT => usize::pow(2, 20), // todo: set limit
+        }
+    }
+
+    /// Returns the min size for a transaction type.
+    pub const fn min_encoded_tx_length(&self) -> usize {
```

Comment:
now we need a few limits here

eip4844 should make use of the blob constants and the others TX_MAX_SIZE + some overhead I think

Reply:
@Rjected where my EIP encyclopaedias at? @shekhirin benchmarks?

Reply:
includes `ReputationChangeKind::BadAnnouncement68`. maybe set all these in a separate pr?

Reply:
let's set this separately, yes.

but I want some notes here that this is just a hint